packages/ubuntu_desktop_installer/test/product_info_extractor_test.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:ubuntu_desktop_installer/utils/product_info_extractor.dart';
+
+void main() {
+  group('ProductInfoExtractor', () {
+    late FileSystem fileSystem;
+    ProductInfoExtractor? productInfoExtractor;
+    setUp(() {
+      fileSystem = MemoryFileSystem();
+
+      productInfoExtractor = ProductInfoExtractor(fileSystem: fileSystem);
+    });
+
+    tearDown(() {
+      productInfoExtractor?.resetState();
+    });
+
+    test('should return product info when iso file exists', () async {
+      final isoPath = '/cdrom/.disk/info';
+
+      await fileSystem.file(isoPath).create(recursive: true).then((File f) {
+        f.writeAsString(
+            'Ubuntu 21.04 "Hirsute Hippo" - Release amd64 (20210420)');
```

Comment:
It would be useful to test with the contents of various ISOs (re [your comment](https://github.com/canonical/ubuntu-desktop-installer/pull/119#discussion_r665750514)).